packages/pdf/src/RPA/PDF/keywords/document.py

```diff
@@ -0,0 +1,510 @@
+
+    def __init__(self, ctx):
+        super().__init__(ctx)
+        self.fpdf = PDF()
+        self._output_directory = Path(".")
+
+    @property
+    def output_directory(self):
+        return self._output_directory
+
+    @output_directory.setter
+    def output_directory(self, path: str):
+        self.output_directory = Path(path)
+
+    def __del__(self):
+        self.close_all_pdf_documents()
+
+    def close_all_pdf_documents(self) -> None:
+        """Close all opened PDF file descriptors."""
+        for filename, fileobject in self.ctx.fileobjects.items():
+            fileobject.close()
+            self.logger.debug('PDF "%s" closed', filename)
+        self.ctx.anchor_element = None
+        self.ctx.fileobjects = {}
+        self.ctx.active_pdf = None
+        self.ctx.active_fileobject = None
+        self.ctx.active_fields = None
+        self.ctx.rpa_pdf_document = None
+
+    @keyword
+    def open_pdf_document(self, source_pdf: str = None) -> None:
+        """Open PDF document.
+
+        :param source_pdf: filepath to the source pdf
+        :raises ValueError: if PDF is already open
+
+        Also opens file for reading.
+        """
+        if source_pdf is None:
+            raise ValueError("Source PDF is missing")
+        if str(source_pdf) in self.ctx.fileobjects.keys():
+            raise ValueError(
+                "PDF file is already open. Please close it before opening again."
+            )
+        self.ctx.active_pdf = str(source_pdf)
+        self.ctx.active_fileobject = open(source_pdf, "rb")
+        self.ctx.active_fields = None
+        self.ctx.fileobjects[source_pdf] = self.ctx.active_fileobject
+        self.ctx.rpa_pdf_document = None
+
+    @keyword
+    def html_to_pdf(
+        self,
+        content: str = None,
+        target_pdf: str = None,
+        variables: dict = None,
+        create_dirs: bool = True,
+        exists_ok: bool = True,
+    ) -> None:
+        """Use HTML content to generate PDF file.
+
+        :param content: HTML content
+        :param target_pdf: filepath where to save PDF document
+        :param variables: dictionary of variables to fill into template, defaults to {}
+        :param create_dirs: directory structure is created if it is missing,
+         default `True`
+        :param exists_ok: file is overwritten if it exists, default `True`
+        """
+        required_param([content, target_pdf], "html_to_pdf")
+        variables = variables or {}
+
+        html = content
+
+        for key, value in variables.items():
+            html = html.replace("{{" + key + "}}", str(value))
+
+        default_output = Path(self.output_directory / "html2pdf.pdf")
+        output_filepath = Path(target_pdf) if target_pdf else default_output
+
+        self._write_html_to_pdf(html, output_filepath, create_dirs, exists_ok)
+
+    def _write_html_to_pdf(self, html, output_path, create_dirs, exists_ok):
+        # TODO: is this needed?
```

Comment:
Figure out TODO, obviously. IMO it could just always overwrite and always create missing parent dirs.